### PR TITLE
feat: Add sourcemaps resolve command

### PR DESCRIPTION
**Pull request by kamilogorek**
Based on https://github.com/getsentry/sentry-cli/pull/1181 so it needs to get merged first.

Includes implementation and integration tests. Only the last commit of this PR is relevant.

### CODE REVIEW

File: src/commands/sourcemaps/resolve.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            Arg::new("column")
+                .long("column")
+                .short('c')
+                .value_name("COLUMN")
+                .validator(validate_int)
+                .help("Column number for minified source."),
+        )
+}
+
+/// Retrurns the zero indexed position from matches
```
Comment (AbhiPrasad) [inline src/commands/sourcemaps/resolve.rs:36]:
```suggestion
/// Returns the zero indexed position from matches
```

**Review by AbhiPrasad — APPROVED** [review]

File: src/commands/sourcemaps/resolve.rs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    let sourcemap_path = matches
+        .value_of("path")
+        .ok_or_else(|| format_err!("Sourcemap not provided"))?;
+
+    let sm = sourcemap::decode_slice(&fs::read(PathBuf::from(sourcemap_path))?)?;
+
+    let ty = match sm {
+        DecodedMap::Regular(..) => "regular",
+        DecodedMap::Index(..) => "indexed",
+        DecodedMap::Hermes(..) => "hermes",
```
Comment (AbhiPrasad) [inline src/commands/sourcemaps/resolve.rs:91]: do we just make returning the type a method on `DecodedMap` so we don't have to come back and update this match statement when we add new one's in the future?

File: src/commands/sourcemaps/resolve.rs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    }
+    if let Some(source) = token.get_source() {
+        println!("  source file: {:?}", source);
+    } else {
+        println!("  source file: not found");
+    }
+    println!("  source line: {}", token.get_src_line());
+    println!("  source column: {}", token.get_src_col());
+    println!("  minified line: {}", token.get_dst_line());
+    println!("  minified column: {}", token.get_dst_col());
```
Comment (AbhiPrasad) [inline src/commands/sourcemaps/resolve.rs:62]: what's with the spaces at the start of lines here?

File: src/commands/sourcemaps/resolve.rs (hunk 2 above)
Comment (kamilogorek) [reply src/commands/sourcemaps/resolve.rs:91]: We'd have to change that directly in `rust-sourcemap`, but it changes so rarely that I'd not bother tbh.

File: src/commands/sourcemaps/resolve.rs (hunk 3 above)
Comment (kamilogorek) [reply src/commands/sourcemaps/resolve.rs:62]: It was meant to make the distinction between the lookup input and output more clear:

image

Comment (kamilogorek) [reply src/commands/sourcemaps/resolve.rs:74]: Also added better indent formatting (code ugly af, but it works :p)

image

**Pull request merged**